22_pipeslab/pipeslab.txt

```diff
@@ -1,21 +1,23 @@
 For each of the following, type a single command line that performs:
 
   1. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner.
-  Answer: 
+  Answer: ls -l | cut -c4 | grep x | wc | tr -s ' ' | cut -d" " -f2
 
   2. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner, group and others.
-  Answer: 
+  Answer: ls -l | cut -c4,7,10 | grep xxx | wc | tr -s ' ' | cut -d" " -f2
 
   3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
-  Answer: 
+  Answer: ls -l --time-style=full-iso| grep $(date +%Y-%m-%d) | wc | tr -s ' ' | cut -d" " -f2
 
   4. Modify (1) and (2) to count files only (no directories)
   Answer:
+  ls -l | cut -c1,4 | grep -x | wc | tr -s ' ' | cut -d" " -f2
+  ls -l | cut -c1,4,7,10 | grep -xxx | wc | tr -s ' ' | cut -d" " -f2
 
   5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
-
+/sbin/ifconfig | grep "inet " | grep -v 127.0.0.1 | tr -s ' ' | cut -d" " -f3
 6. Joe wanted to sort a text file in place so he tried to run the command "sort myfile.txt > myfile.txt". What happened to the file? Why?
-
+The content was deleted. I'm not sure what is the reason. Maybe sharing violation (trying writing while reading).
```

Comment:
The reason is actually much simpler - the shell will process the redirection first, and by the time ```sort``` is running the file is already empty..